fabio/edfimage.py

```diff
@@ -255,27 +313,51 @@ def _extract_header_metadata(self, capsHeader=None):
             self._data_compression = None
 
         bpp = self._dtype.itemsize
-        calcsize *= bpp
+        calcsize = counts*bpp
 
-        # Checks can be done only for uncompressed data
-        if self._data_compression is None:
-            if self.blobsize is None:
+        # only if blobsize is None it can be replaced with calcsize
+        if self.blobsize is None:
+            if self._data_compression is None:
                 # In some edf files the blobsize is not written.
                 # For uncompressed data it can be set to the calculated size.
                 self.blobsize = calcsize
 
-            elif self.blobsize < calcsize:
-                # The EDF binary block can store up to self.blobsize bytes.
-                # If the required blobsize is smaller, there is no
-                # problem and the data can be stored, otherwise if the available blobsize is too small the data must be
-                # truncated
-                logger.warning("Malformed file. The specified size of the data block is smaller than the expected size (%i < %i). This frame and the following could be broken.", self.size, calcsize)
-                self.size = calcsize
+        # PB38k20190607: logger warning for self.blobsize < calcsize removed,
+        #        because blobsize can be 0 when binary data are read from
+        #        an external file (to be added later).
+        #        The check must be done when actually reading the data.
 
         if self.size is None:
-            # Preset with the calculated size, will be updated after decompressing
+            # preset with the calculated size, will be updated later
+            # with a better value when it becomes available, e.g.
+            # after decompression
             self.size = calcsize
 
+        #+++++++++++++++++++++++++++++
+        # PB38k20190607: ATTENTION, weird!:
+        # little_endian^=LowByteFirst, big_endian^=HighByteFirst
+        # Why should _data_swap_needed depend on bpp?
```

Comment:
Hi Peter,
The code is not the right place for discussing implementation details. Please use the pull-request facility for this in order to keep the code clean an tidy.

About this topic, little- & big-endian differ only if the datatype has a length longer then 1